test/parallel_api/dynamic_selection/sycl/test_auto_tune_policy_sycl_profile.pass.cpp

```diff
@@ -0,0 +1,551 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <iostream>
+#include <thread>
+#include "oneapi/dpl/dynamic_selection"
+#include "support/test_dynamic_selection_utils.h"
+#include "support/test_config.h"
+#if TEST_DYNAMIC_SELECTION_AVAILABLE
+#    include "support/sycl_sanity.h"
+
+template <bool call_select_before_submit, typename Policy, typename UniverseContainer>
+int
+test_auto_submit_wait_on_event(UniverseContainer u, int best_resource)
+{
+    using my_policy_t = Policy;
+
+    int j;
+    std::vector<double> v(1000000, 0.0);
+
+    my_policy_t p{u};
+    auto n_samples = u.size();
+
+    const int N = 10;
+    std::atomic<int> ecount = 0;
+    bool pass = true;
+
+    for (int i = 1; i <= N; ++i)
+    {
+        if (i <= 2 * n_samples && (i - 1) % n_samples != best_resource)
+        {
+            j = 100;
+        }
+        else
+        {
+            j = 0;
+        }
+        // we can capture all by reference
+        // the inline_scheduler reports timings in submit
```

Comment:
These comments seems incorrect, since this is not the inline_scheduler.

Reply:
Fixed